parser/src/java/org/apache/hadoop/hive/ql/parse/IdentifiersParser.g

```diff
@@ -464,6 +473,7 @@ atomExpression
     | whenExpression
     | (subQueryExpression)=> (subQueryExpression)
         -> ^(TOK_SUBQUERY_EXPR TOK_SUBQUERY_OP subQueryExpression)
+    | (functionName LPAREN (leading=KW_LEADING | trailing=KW_TRAILING | KW_BOTH)? (trim_characters=selectExpression)? KW_FROM (str=selectExpression) RPAREN) => trimFunction
```

Comment:
I think this would match any function name, that then would be converted to `trim` (or any other variant) by the rules in `trimFunction`?

I think one way of fixing this could be to declare `TRIM` as a keyword, and match directly on `TRIM` (TRIM is a reserved keyword in SQL standard) rather than `functionName`.

Could you also add a negative test, i.e., a test where the trim construct is correct but the function name is different, and make sure it fails and it is not transformed by the parser?

Reply:
this is a syntetic predicate expression; narrowing it down furthere will help for sure ; but since `trimFunction` also matches the `functionName LPAREN` prefix I think it should be placed in the `function` rule